ndb_client/ndb_client.go

```diff
@@ -44,39 +51,36 @@ func NewNDBClient(username, password, url, caCert string, skipVerify bool) *NDBC
 	return &NDBClient{username, password, url, client}
 }
 
-func (ndbClient *NDBClient) Get(path string) (*http.Response, error) {
-	url := ndbClient.url + "/" + path
-	req, err := http.NewRequest(http.MethodGet, url, nil)
-	if err != nil {
-		// fmt.Println(err)
-		return nil, err
+func (ndbClient *NDBClient) NewRequest(method, endpoint string, requestBody interface{}) (*http.Request, error) {
+
+	url := ndbClient.url + "/" + endpoint
+
+	var body io.Reader
+
+	if requestBody != nil {
+		// Serialize the request body to JSON.
+		payload, err := json.Marshal(requestBody)
+		if err != nil {
+			return nil, err
+		}
+		// Create a reader from the serialized payload.
+		body = bytes.NewReader(payload)
 	}
-	req.SetBasicAuth(ndbClient.username, ndbClient.password)
-	return ndbClient.client.Do(req)
-}
 
-func (ndbClient *NDBClient) Post(path string, body interface{}) (*http.Response, error) {
-	url := ndbClient.url + "/" + path
-	payload, _ := json.Marshal(body)
-	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(payload))
+	// Create a new HTTP request with the specified method and URL.
+	req, err := http.NewRequest(method, url, body)
 	if err != nil {
-		// fmt.Println(err)
 		return nil, err
 	}
+
+	// Set headers
```

Comment:
optional; setting headers itself can be a separate method.

Reply:
We can do that once we have more headers to set, I think it should be fine for just 2 headers for now? What do you think?